src/JBBuybackHook.sol

```diff
@@ -465,16 +521,53 @@ contract JBBuybackHook is JBPermissioned, IJBBuybackHook {
         // Add the amount to mint to the leftover mint amount.
         partialMintTokenCount += mulDiv(amountToMintWith, context.weight, weightRatio);
 
-        // Mint the calculated amount of tokens for the beneficiary, including any leftover amount.
+        // Mint the total amount of tokens to be vested to this contract.
         // This takes the reserved rate into account.
         // slither-disable-next-line unused-return
-        CONTROLLER.mintTokensOf({
+        uint256 amountToVest = CONTROLLER.mintTokensOf({
             projectId: context.projectId,
             tokenCount: exactSwapAmountOut + partialMintTokenCount,
-            beneficiary: address(context.beneficiary),
+            beneficiary: address(this),
             memo: "",
             useReservedPercent: true
         });
+
+        // Get a reference to the project's token.
+        IJBToken token = CONTROLLER.TOKENS().tokenOf(context.projectId);
+
+        // Add the calculated amount of tokens to be vested for the beneficiary, including any leftover amount.
+        // This takes the reserved rate into account.
+        // slither-disable-next-line unused-return
+        _vestingBuybacksFor[token][context.beneficiary].push(
```

Comment:
I am a little bit worried about this approach. 
Anyone can specify a `beneficiary`, when buying tokens in project. This means that malicious actor can pay dust amounts in different transactions only to fill victim's `_vestingBuybacksFor` array with many entries. If the beneficiary then have solid `JBVestingBuyback` that we have paid for, he should every time pay the gas to iterate over all dummy entries, when calling `claimVestedBuybacksFor`. 

A potential fix is to introduce function `claimVestedBuybackForIndex`, so user can specify an id from `_vestingBuybacksFor` to claim the tokens. This way the "attack" won't have an impact for the "victim".

Reply:
great idea. ill post an update in a bit

Reply:
i pushed an update that adds `startIndex` as an arg to `claimVestedBuybackForIndex`. i also removed the storage of indexes since its no longer needed, and removed the view fn since front end can easily compute from events.